src/api/reports/views.py

```diff
@@ -0,0 +1,109 @@
+import os
+from datetime import datetime
+
+from django.http import FileResponse
+from django_countries import countries
+from rest_framework.exceptions import ParseError
+from rest_framework.views import APIView
+
+from . import species_report
+from . import stats
+from ..models import Species
+from ..resources.base import UserCountryPermission
+
+
+class SpeciesReportView(APIView):
+    permission_classes = [UserCountryPermission]
+    EXCEL_MIME_TYPE = (
+        "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
+    )
+    table_data_order = [
+        "conservation_landscape_stats",
+        "fragment_landscape_stats",
+        "restoration_landscape_stats",
+        "survey_landscape_stats",
+    ]
+
+    def get_file_name(self, ext="xlsx"):
+        date = str(datetime.now().date())
+        return f"species-report-{date}.{ext}"
+    
+    def _parse_query_params(self, request):
+        country_code = request.query_params.get("country")
+        date = request.query_params.get("date")
+        try:
+            species_id = int(request.query_params.get("species"))
+        except (TypeError, ValueError):
+            species_id = None
+        
+        if not country_code:
+            raise ParseError("Missing country")
+
+        if not date:
+            raise ParseError("Missing date")
+        
+        if not species_id:
+            raise ParseError("Missing species")
+
+        return country_code, date, species_id
+    
+    def get_report_data(self, country_code, date, species_id):
+        try:
+            species = Species.objects.get(id=species_id)
+            species_name = species.full_name
+        except Species.DoesNotExist:
+            species_name = ""        
+        landscape_stats = stats.calc_landscape_stats(country_code, date, species_id)
+        country_name = dict(countries).get(country_code)
+        table_data = []
+        total_num_landscapes = 0
+        total_habitat_area = 0
+        total_protected_area = 0
+        total_percent_protected_area = None
+
+        for k in self.table_data_order:
+            lss = landscape_stats[k]
+            num_landscapes = lss.get("num_landscapes")
+            habitat_area = lss.get("habitat_area")
+            percent_protected_area = lss.get("percent_protected_area")
+
+            total_num_landscapes += num_landscapes
+            total_habitat_area += habitat_area
+            total_protected_area += lss.get("protected_area")
+
+            table_data.append([
+                num_landscapes,
+                habitat_area,
+                percent_protected_area,
+            ])
+        
+        if total_habitat_area:
```

Comment:
Should protect against dividing by 0

Reply:
if `total_habitat_area` is 0 it won't go in the if statement